mirrord/agent/src/steal/ip_tables/output.rs

```diff
@@ -20,12 +20,18 @@ where
 {
     const ENTRYPOINT: &'static str = "OUTPUT";
 
-    pub fn create(ipt: Arc<IPT>, chain_name: String) -> Result<Self> {
+    pub fn create(ipt: Arc<IPT>, chain_name: String, pod_ips: Option<&str>) -> Result<Self> {
         let managed = IPTableChain::create(ipt, chain_name)?;
 
+        let exclude_source_ips = pod_ips
+            .map(|pod_ips| format!("! -s {pod_ips}"))
+            .unwrap_or_default();
```

Comment:
Wait, so the complete rule is now like "ignore TCP packets sent from agent's GID, unless they're sent from pod's IPs"? How does it help? Isn't this part a noop? :eyes:

Reply:
Not quite because the outgoing handler runs in the target process namespace so the request do originate from the agent (checking what happens on ephemerals)

Reply:
ephemeral agent works just fine